### PR TITLE
APMSP-1241 Directly import trace-agent stats code for client-side stats

### PR DESCRIPTION
**Pull request by ajgajg1134**
**WIP: This PR should be updated to point to the released `v0.58` version of the trace-agent when it is available and then should be ready for merge. However this code is very unlikely to change other than changing the go.mod version so it should be safe to review now with minimal disruption.**

### What does this PR do?
Begin using the trace-agent's Concentrator instead of our own (mostly copied) implementation from the trace-agent. 

Unfortunately there's some additional changes here needed like updating to go 1.22 as that's the minimum supported version of the trace-agent (but this is aligned with dd-trace-go's published version support policy). Additionally the google.golang.org/api contrib needed to be updated to support the more recent versions required by the trace-agent. I believe these imports / upgrades are _ok_ but if we need to target a smaller import and modularize some more of the trace-agent to reduce impact here just let me know!

### Motivation
Some new features introduced in the trace-agent concentrator have been missed here over time and this brings back feature parity when using client side stats (e.g. peer tags aggregation, marking trace root spans). We will eventually be enabling client stats by default (as it's more efficient!) and this is a required step to ensure that we don't break important stats features.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,14 +1,18 @@
 module gopkg.in/DataDog/dd-trace-go.v1
 
-go 1.21
+go 1.22.0
+
+toolchain go1.22.5
```
Comment (eliottness) [inline go.mod:5]: I believe adding toolchain directives in our repo is one of the things we agreed to avoid
```suggestion
```

**Review by eliottness — APPROVED** [review]
Approving on behalf of appsec only for appsec code. Thanks for starting out the migration to rolling out go 1.21 👍

File: go.mod (hunk 1 above)
Comment (ajgajg1134) [reply go.mod:5]: Oops, not sure what tool or command I ran that added these, I'll remove them

Comment (nsrip-dd) [reply go.mod:5]: I've wondered the same thing. It happens when the `go get`/`go mod` command updates the `go` directive due some dependency requiring the newer release. If you manually change the `go` directive yourself before doing any dependency changes, the `toolchain` directive won't be added. Ref: https://go.dev/ref/mod#go-mod-file-toolchain

**Comment by darccio** [comment]
@ajgajg1134 Was the agent released? When will this able to be merged?

**Review by darccio — APPROVED** [review]

**Pull request merged**